dotnet/src/Connectors/Connectors.GoogleVertexAI/Extensions/VertexAIServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,176 @@
+        return builder;
+    }
+
+    /// <summary>
+    /// Adds Vertex AI Gemini Text Generation service to the specified service collection.
+    /// </summary>
+    /// <param name="services">The service collection to add the Gemini Text Generation service to.</param>
+    /// <param name="modelId">The model for text generation.</param>
+    /// <param name="apiKey">The API key for authentication Gemini API.</param>
+    /// <param name="location">The region to process the request</param>
+    /// <param name="projectId">Your project ID</param>
+    /// <param name="serviceId">Optional service ID.</param>
+    /// <returns>The updated service collection.</returns>
+    public static IServiceCollection AddVertexAIGeminiTextGeneration(
+        this IServiceCollection services,
+        string modelId,
+        string apiKey,
+        string location,
+        string projectId,
+        string? serviceId = null)
+    {
+        Verify.NotNull(services);
+        Verify.NotNull(modelId);
+        Verify.NotNull(apiKey);
+        Verify.NotNull(location);
+        Verify.NotNull(projectId);
+
+        return services.AddKeyedSingleton<ITextGenerationService>(serviceId, (serviceProvider, _) =>
+            new VertexAIGeminiTextGenerationService(
+                model: modelId,
+                apiKey: apiKey,
+                location: location,
+                projectId: projectId,
+                httpClient: HttpClientProvider.GetHttpClient(serviceProvider),
+                loggerFactory: serviceProvider.GetService<ILoggerFactory>()));
+    }
+
+    /// <summary>
+    /// Adds Vertex AI Gemini Chat Completion and Text Generation services to the kernel builder.
+    /// </summary>
+    /// <param name="builder">The kernel builder.</param>
+    /// <param name="modelId">The model for text generation.</param>
+    /// <param name="apiKey">The API key for authentication Gemini API.</param>
+    /// <param name="location">The region to process the request</param>
+    /// <param name="projectId">Your project ID</param>
+    /// <param name="serviceId">The optional service ID.</param>
+    /// <param name="httpClient">The optional custom HttpClient.</param>
+    /// <returns>The updated kernel builder.</returns>
+    public static IKernelBuilder AddVertexAIGeminiChatCompletion(
+        this IKernelBuilder builder,
+        string modelId,
+        string apiKey,
+        string location,
+        string projectId,
+        string? serviceId = null,
+        HttpClient? httpClient = null)
+    {
+        Verify.NotNull(builder);
+        Verify.NotNull(modelId);
+        Verify.NotNull(apiKey);
+        Verify.NotNull(location);
+        Verify.NotNull(projectId);
+
+        builder.Services.AddKeyedSingleton<IChatCompletionService>(serviceId, (serviceProvider, _) =>
+            new VertexAIGeminiChatCompletionService(
+                model: modelId,
+                apiKey: apiKey,
+                location: location,
```

Comment:
nit.
As the description says, location is the region, suggest using region argument on the extensions.

```suggestion
                region: region,
```